### DISCUSSION TITLE
Using a CSS variable to set the colums in a grid

### DISCUSSION THREAD

**Discussion by AaronPresley** [tailwindlabs/tailwindcss#12397] [Help] [answered]
Hello,

I feel like this is an easy one, but I haven't been able to find a direct answer to my question.

I'm unable to use a CSS variable to set the number of columns in a grid, code below and  [demo here](https://play.tailwindcss.com/U6pyVzdsIM).

```html
<div class="bg-gray-100 p-8">
  <div class="grid grid-cols-[--count] bg-[--color]" style="--count:3; --color:pink;">
    <div>Col Here</div>
    <div>Col Here</div>
    <div>Col Here</div>
  </div>
</div>
```

In this example, the `--color` variable correctly sets the background, but the `--count` variable has no effect on the grid columns. Would you please let me know what I'm missing here?

Thanks for your time 🙏

**Comment by wongjn** [comment, marked as answer]
You are missing a valid value to pass to `grid-cols-[]`. `grid-cols-*` corresponds to the `grid-template-columns` CSS property, of which, `3` is an invalid value. To emulate `grid-cols-3` for `--count:3`, you'd want the arbitrary value to be `repeat(var(--count),1fr)`. See it working in [this Tailwind Play](https://play.tailwindcss.com/EXr64Fzv6n)

**Reply by AaronPresley** [reply]
Thank you so much @wongjn - that did the trick! I clearly have more to learn regarding CSS variables 🙂
